Automattic/portkey-android
# Normalize exported images to 9:16

Fix #451 

Title has it, in this PR we added method `normalizeHeightExportTo916()` to ensure the `clonedViewSpecs` adjust the measures to 9:16.

This is how it works:
1. if the screen is 16:9, we're OK and return as usual
2. if not, we resolve what the height should be for the screen's given width.
3. if the result of (2) is less than actual height, set the new target height for the cloned view to the number calculated as of (2) (cropping)
4. if the result of (2) is greater than actual height --> crop the width to match the ratio needed for the given height

Step (4) here is less (never?) likely to happen in practice, given most popular screens are some sort of not-square device where the width is always less than height (portrait mode), but it's coded so we make sure we have a defined state for all cases.

For example, using a Pixel 3 with a screen of 1440x2792:

**Before**:
produces an image of 1440x2792 (matching the screen's resolution)
https://testmzorz3.files.wordpress.com/2020/07/wp-1595874869618.jpg


**After**:
produces an image that is now 1440x2560 (matching the target ratio)
https://testmzorz3.files.wordpress.com/2020/07/wp-1595874869642.jpg

Will continue testing and add here.

To test:
1. take a phone that is not 16:9
2. install portkey and produce an slide with an image background
3. verify the output image is 16:9
